src/OrleansProviders/Streams/Common/SimpleCache/SimpleQueueCache.cs

```diff
@@ -79,12 +79,7 @@ public SimpleQueueCache(int cacheSize, Logger logger)
         /// </summary>
         public virtual bool IsUnderPressure()
         {
-            if (cachedMessages.Count == 0) return false; // empty cache
-            if (Size < maxCacheSize) return false; // there is still space in cache
-            if (cacheCursorHistogram.Count == 0) return false;    // no cursors yet - zero consumers basically yet.
-            // cache is full. Check how many cursors we have in the oldest bucket.
-            int numCursorsInLastBucket = cacheCursorHistogram[0].NumCurrentCursors;
-            return numCursorsInLastBucket > 0;
+            return cacheCursorHistogram.Count >= NUM_CACHE_HISTOGRAM_BUCKETS;
```

Comment:
Can you provide your reasoning why the previous IsUnderPressure logic was better?
I don't consider minimizing the changes as a high priority.  I'd prioritize simplicity and maintainability, and am of the opinion that the code is now simpler and more maintainable.
While figuring out what the problem was, understanding the importance of each check (within IsUnderPressure) added to cognitive load.  Once I understood the conditions and realized the checks were not necessary, I saw no reason to keep them.